src/Files.App/Data/Models/ItemViewModel.cs

```diff
@@ -937,117 +937,142 @@ private async Task<BitmapImage> GetShieldIcon()
+		/// <param name="item"></param>
+		/// <returns></returns>
+		private async Task LoadThumbnailAsync(ListedItem item)
+		{
+			// Cancel if thumbnails aren't enabled
+			var thumbnailSize = folderSettings.GetRoundedIconSize();
+			if (UserSettingsService.FoldersSettingsService.ShowThumbnails == false || thumbnailSize < 48)
+				return;
+
+			// Get thumbnail
+			var icon = await FileThumbnailHelper.GetIconAsync(
+					item.ItemPath,
+					thumbnailSize,
+					item.IsFolder,
+					false,
+					IconOptions.None);
 
-				var iconOverlay = await FileThumbnailHelper.GetIconOverlayAsync(item.ItemPath, false);
-				if (iconOverlay is not null)
+			if (icon.IconData is not null)
+			{
+				await dispatcherQueue.EnqueueOrInvokeAsync(async () =>
 				{
-					// Assign the icon overlay to the listed item
-					await dispatcherQueue.EnqueueOrInvokeAsync(async () =>
-					{
-						item.IconOverlay = await iconOverlay.ToBitmapAsync();
-						item.ShieldIcon = await GetShieldIcon();
-					}, Microsoft.UI.Dispatching.DispatcherQueuePriority.Low);
-				}
+					// Assign FileImage property
+					var image = await icon.IconData.ToBitmapAsync();
+					if (image is not null)
+						item.FileImage = image;
+				}, Microsoft.UI.Dispatching.DispatcherQueuePriority.Low);
 			}
-			else
+		}
+
+		/// <summary>
+		/// Tries getting a cached thumbnail from the system. This is usually only needed for cloud locations, otherwise LoadThumbnailAsync does the job.
+		/// </summary>
+		/// <param name="item"></param>
+		/// <returns></returns>
+		private async Task LoadCachedThumbnailAsync(ListedItem item)
+		{
+			// Cancel for exe files which don't need thumbnails
+			if (item.IsExecutable)
+				return;
+
+			// Cancel if thumbnails aren't enabled
+			var thumbnailSize = folderSettings.GetRoundedIconSize();
+			if (UserSettingsService.FoldersSettingsService.ShowThumbnails == false || thumbnailSize < 48)
+				return;
+
+			// Get icon
+			var icon = await FileThumbnailHelper.GetIconAsync(
+					item.ItemPath,
+					thumbnailSize,
+					item.IsFolder,
+					true,
+					IconOptions.None);
+
+			// Loop until cached thumbnail is loaded or timeout is reached
+			var cancellationTokenSource = new CancellationTokenSource(3000);
+			while (!icon.isIconCached)
 			{
-				var getIconOnly = UserSettingsService.FoldersSettingsService.ShowThumbnails == false || thumbnailSize < 48;
-				var iconInfo = await FileThumbnailHelper.GetIconAsync(
+				icon = await FileThumbnailHelper.GetIconAsync(
 					item.ItemPath,
 					thumbnailSize,
+					item.IsFolder,
 					true,
-					false, getIconOnly ? IconOptions.ReturnIconOnly : IconOptions.None);
+					IconOptions.None);
 
-				if (iconInfo.IconData is not null)
-				{
-					await dispatcherQueue.EnqueueOrInvokeAsync(async () =>
-					{
-						item.FileImage = await iconInfo.IconData.ToBitmapAsync();
-					}, Microsoft.UI.Dispatching.DispatcherQueuePriority.Low);
-				}
+				if (cancellationTokenSource.Token.IsCancellationRequested)
```

Comment:
```suggestion
				if (icon.isIconCached || cancellationTokenSource.Token.IsCancellationRequested)
```
This prevents an extra 500 msec wait.